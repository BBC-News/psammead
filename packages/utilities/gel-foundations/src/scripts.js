export latin from './scripts/latin-and-cyrillic';
export cyrillic from './scripts/latin-and-cyrillic';
export latinDiacritics from './scripts/latin-with-diacritics';
export arabic from './scripts/arabic';
export arabicPashto from './scripts/arabic-pashto';
export bengali from './scripts/bengali';
export burmese from './scripts/burmese';
export hindi from './scripts/devanagari-and-gurmukhi';
export nepali from './scripts/devanagari-and-gurmukhi';
export sinhalese from './scripts/sinhalese';
export tamil from './scripts/tamil';
export thai from './scripts/thai';
export chinese from './scripts/no-ascenders-or-descenders';
export korean from './scripts/no-ascenders-or-descenders';
export ethiopic from './scripts/no-ascenders-or-descenders';
