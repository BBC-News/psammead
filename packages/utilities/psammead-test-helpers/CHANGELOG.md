# Psammead Test Helpers Changelog

| Version | Description |
|---------|-------------|
| 0.3.0   | [PR#224](https://github.com/BBC-News/psammead/pull/224) Adds a helper method for testing package export values and uses it to test its own exports. |
| 0.2.1   | [PR#255](https://github.com/BBC-News/psammead/pull/255) Bumps package version to fix what appears to be an erroneously-published version. |
| 0.1.2   | [PR#212](https://github.com/BBC-News/psammead/pull/212) Update package description and README. |
| 0.1.1   | [PR#173](https://github.com/BBC-News/psammead/pull/173) Update PRs welcome link |
| 0.1.0   | [PR#80](https://github.com/BBC-News/psammead/pull/80) Create initial package, pulled in from [simorgh](https://github.com/BBC-News/simorgh). |
