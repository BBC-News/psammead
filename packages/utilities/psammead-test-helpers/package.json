{
  "name": "@bbc/psammead-test-helpers",
  "version": "0.3.0",
  "main": "dist/index.js",
  "description": "A collection of helper methods for implementing Jest snapshot tests for styled-components, required by many Psammead components.",
  "repository": {
    "type": "git",
    "url": "https://github.com/BBC-News/psammead/tree/latest/packages/utilities/psammead-test-helpers"
  },
  "author": {
    "name": "Psammead Maintainers",
    "email": "PsammeadMaintainers@bbc.co.uk"
  },
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/bbc/psammead/issues"
  },
  "homepage": "https://github.com/BBC-News/psammead/blob/latest/packages/utilities/psammead-test-helpers/README.md",
  "keywords": [
    "bbc",
    "utilities",
    "test",
    "helpers",
    "jest",
    "snapshot"
  ],
  "dependencies": {
    "jest-styled-components": "^6.3.1",
    "react-test-renderer": "^16.6.3"
  }
}
