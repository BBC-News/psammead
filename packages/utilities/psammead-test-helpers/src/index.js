import renderer from 'react-test-renderer';
import 'jest-styled-components';
import ShallowRenderer from 'react-test-renderer/shallow';

export const shouldMatchSnapshot = (title, component) => {
  it(title, () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
};

export const shallowRender = component => {
  const shallowRenderer = new ShallowRenderer();
  shallowRenderer.render(component);
  const result = shallowRenderer.getRenderOutput();

  return result;
};

export const shouldShallowMatchSnapshot = (title, component) => {
  it(title, () => {
    const tree = shallowRender(component);
    expect(tree).toMatchSnapshot();
  });
};

export const isNull = (title, component) => {
  it(title, () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toBeNull();
  });
};

const errorIfMissingKey = (keys, object, message) => {
  keys.forEach(key => {
    if (!(key in object)) {
      throw new Error(`Missing value '${key}' in ${message}.`);
    }
  });
};

const checkKeysExistInBothObjects = (object1, object2, message1, message2) => {
  const object1Keys = Object.keys(object1);
  const object2Keys = Object.keys(object2);

  errorIfMissingKey(object1Keys, object2, message2);
  errorIfMissingKey(object2Keys, object1, message1);
};

const checkTypesOfExports = (
  actualExportsByName,
  actualExports,
  expectedExports,
  utilityName,
) => {
  actualExportsByName.forEach(actualExportName => {
    const actualExportValue = actualExports[utilityName][actualExportName];
    const expectedExport = expectedExports[utilityName][actualExportName];
    const typeCheck = typeof actualExportValue === expectedExport; // eslint-disable-line valid-typeof

    // if this fails it is likely that an export is missing from the unit test expectation
    expect(typeCheck).toBe(true);
  });
};

export const testUtilityPackages = (
  actualExports,
  expectedExports,
  packageName,
) => {
  const actualUtilities = Object.keys(actualExports);

  // check if the actual exported file has defined expected values to match and that the expected values are actually exported
  checkKeysExistInBothObjects(
    actualExports,
    expectedExports,
    `the actual utilities for '${packageName}'`,
    `the expected utilities for '${packageName}'`,
  );

  actualUtilities.forEach(utilityName => {
    const actualExportsByName = Object.keys(actualExports[utilityName]);

    // check if each of the actual exported consts have an expected value to match and that the expected values are actually exported
    checkKeysExistInBothObjects(
      actualExports[utilityName],
      expectedExports[utilityName],
      `the actual export for '${packageName}/${utilityName}'`,
      `the expected export for '${packageName}/${utilityName}'`,
    );

    checkTypesOfExports(
      actualExportsByName,
      actualExports,
      expectedExports,
      utilityName,
    );
  });
};
