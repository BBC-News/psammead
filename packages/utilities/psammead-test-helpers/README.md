# psammead-test-helpers &middot; [![GitHub license](https://img.shields.io/badge/license-Apache%202.0-blue.svg)](https://github.com/BBC-News/psammead/blob/latest/LICENSE) [![npm version](https://img.shields.io/npm/v/@bbc/psammead-test-helpers.svg)](https://www.npmjs.com/package/@bbc/psammead-test-helpers) [![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg)](https://github.com/BBC-News/psammead/blob/latest/CONTRIBUTING.md)

This package is a collection helper methods used for snapshot testing, with support for styled components.

## Installation

```jsx
npm install react react-dom @bbc/psammead-test-helpers --save-dev
```

## Usage

```jsx
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';

shouldMatchSnapshot(
  'should render correctly',
  <h1>Hello World</h1>,
);
```

## Functions

| Name                       | Arguments        | Description  |
|:---------------------------|:-----------------|:-------------|
| shouldMatchSnapshot        | title, component | Renders the component using react-test-renderer, converts it to JSON and asserts that it matches the given snapshot, which will be saved in the `__snapshots__` directory. The first argument `title` is the title for the test. |
| shallowRender              | component        |  Shallow renders the component using react-test-renderer and returns the render output |
| shouldShallowMatchSnapshot | title, component | Shallow renders the component using react-test-renderer and asserts that it matches the given snapshot, which will be saved in the `__snapshots__` directory. The first argument `title` is the title for the test. |
| isNull                     | title, component    | Renders the component using react-test-renderer, converts it to JSON and asserts that it is null. The first argument `title` is the title for the test. |

## Roadmap

There is currently a plan to migrate from `react-test-renderer` to `enzyme` as it provides a nicer higher-level API over the same functionality, as well as having support for more granular unit tests.

## Contributing

Psammead is completely open source. We are grateful for any contributions, whether they be new components, bug fixes or general improvements. Please see our primary contributing guide which can be found at [the root of the Psammead respository](https://github.com/BBC-News/psammead/blob/latest/CONTRIBUTING.md).

### [Code of Conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md)

We welcome feedback and help on this work. By participating in this project, you agree to abide by the [code of conduct](https://github.com/BBC-News/psammead/blob/latest/CODE_OF_CONDUCT.md). Please take a moment to read it.

### License

Psammead is [Apache 2.0 licensed](https://github.com/BBC-News/psammead/blob/latest/LICENSE).
