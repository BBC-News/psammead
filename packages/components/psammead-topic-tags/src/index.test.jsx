import React from 'react';
import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
import { latin, burmese, arabic } from '@bbc/gel-foundations/dist/scripts';
import { TopicTags, TopicTag } from './index';

describe('TopicTags', () => {
  const baseProps = {
    dir: 'ltr',
    service: 'news',
    script: latin,
  };

  shouldMatchSnapshot(
    'should correctly render a single topic for ltr',
    <TopicTags {...baseProps}>
      <TopicTag topicName="test1" topicLink="#" />
    </TopicTags>,
  );

  shouldMatchSnapshot(
    'should correctly render a single topic for rtl',
    <TopicTags {...baseProps} dir="rtl">
      <TopicTag topicName="test1" topicLink="#" />
    </TopicTags>,
  );

  shouldMatchSnapshot(
    'should correctly render multiple topic tags for ltr',
    <TopicTags {...baseProps}>
      <TopicTag topicName="test1" topicLink="#" />
      <TopicTag topicName="test2" topicLink="#" />
      <TopicTag topicName="test3" topicLink="#" />
      <TopicTag topicName="test4" topicLink="#" />
    </TopicTags>,
  );

  shouldMatchSnapshot(
    'should correctly render multiple topic tags for rtl',
    <TopicTags {...baseProps} dir="rtl">
      <TopicTag topicName="test1" topicLink="#" />
      <TopicTag topicName="test2" topicLink="#" />
      <TopicTag topicName="test3" topicLink="#" />
      <TopicTag topicName="test4" topicLink="#" />
    </TopicTags>,
  );

  shouldMatchSnapshot(
    'should correctly render a single topic tag for burmese',
    <TopicTags service="burmese" script={burmese} dir="rtl">
      <TopicTag topicName="test1" topicLink="#" />
    </TopicTags>,
  );

  shouldMatchSnapshot(
    'should correctly render multiple topic tags for arabic',
    <TopicTags service="arabic" script={arabic} dir="rtl">
      <TopicTag topicName="test1" topicLink="#" />
      <TopicTag topicName="test2" topicLink="#" />
      <TopicTag topicName="test3" topicLink="#" />
      <TopicTag topicName="test4" topicLink="#" />
    </TopicTags>,
  );
});
