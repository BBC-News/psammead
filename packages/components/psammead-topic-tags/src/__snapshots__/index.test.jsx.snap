// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicTags should correctly render a single topic for news 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: -0.0625rem;
  margin-bottom: 0;
  margin-left: -0.25rem;
  margin-right: -0.25rem;
  padding: 0;
}

.emotion-2 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.875rem;
  line-height: 1.125rem;
  word-break: break-word;
  margin-top: 0.0625rem;
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-2 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-2 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

.emotion-2 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #222222;
}

.emotion-2 a:hover,
.emotion-2 a:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2 a:visited {
  color: #6E6E73;
}

.emotion-2 a>span {
  background-color: #F2F2F2;
  padding: 0.375rem 0.4375rem;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <a
        href="#"
      >
        <span>
          test1
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`TopicTags should correctly render a single topic tag for burmese 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: -0.0625rem;
  margin-bottom: 0;
  margin-left: -0.25rem;
  margin-right: -0.25rem;
  padding: 0;
}

.emotion-2 {
  font-family: Padauk,Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.875rem;
  line-height: 1.375rem;
  word-break: break-word;
  margin-top: 0.0625rem;
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-2 {
    font-size: 0.875rem;
    line-height: 1.375rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-2 {
    font-size: 0.875rem;
    line-height: 1.375rem;
  }
}

.emotion-2 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #222222;
}

.emotion-2 a:hover,
.emotion-2 a:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2 a:visited {
  color: #6E6E73;
}

.emotion-2 a>span {
  background-color: #F2F2F2;
  padding: 0.375rem 0.4375rem;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <a
        href="#"
      >
        <span>
          test1
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`TopicTags should correctly render a single topic tag for persian 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: -0.0625rem;
  margin-bottom: 0;
  margin-left: -0.25rem;
  margin-right: -0.25rem;
  padding: 0;
}

.emotion-2 {
  font-family: "BBC Reith Qalam",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.875rem;
  line-height: 1.25rem;
  word-break: break-word;
  margin-top: 0.0625rem;
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-2 {
    font-size: 0.875rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-2 {
    font-size: 0.8125rem;
    line-height: 1.25rem;
  }
}

.emotion-2 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #222222;
}

.emotion-2 a:hover,
.emotion-2 a:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2 a:visited {
  color: #6E6E73;
}

.emotion-2 a>span {
  background-color: #F2F2F2;
  padding: 0.375rem 0.4375rem;
}

<div>
  <div
    class="emotion-0 emotion-1"
  >
    <div
      class="emotion-2 emotion-3"
    >
      <a
        href="#"
      >
        <span>
          test1
        </span>
      </a>
    </div>
  </div>
</div>
`;

exports[`TopicTags should correctly render multiple topic tags for arabic 1`] = `
.emotion-0 {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: -0.0625rem;
  margin-bottom: 0;
  margin-left: -0.25rem;
  margin-right: -0.25rem;
  padding: 0;
}

.emotion-2 {
  font-family: "BBC Reith Qalam",Arial,Verdana,Geneva,Helvetica,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.875rem;
  line-height: 1.25rem;
  word-break: break-word;
  margin-top: 0.0625rem;
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-2 {
    font-size: 0.875rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-2 {
    font-size: 0.8125rem;
    line-height: 1.25rem;
  }
}

.emotion-2 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #222222;
}

.emotion-2 a:hover,
.emotion-2 a:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2 a:visited {
  color: #6E6E73;
}

.emotion-2 a>span {
  background-color: #F2F2F2;
  padding: 0.375rem 0.4375rem;
}

<div>
  <ul
    class="emotion-0 emotion-1"
    role="list"
  >
    <li
      class="emotion-2 emotion-3"
    >
      <a
        href="#"
      >
        <span>
          test1
        </span>
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        href="#"
      >
        <span>
          test2
        </span>
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        href="#"
      >
        <span>
          test3
        </span>
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        href="#"
      >
        <span>
          test4
        </span>
      </a>
    </li>
  </ul>
</div>
`;

exports[`TopicTags should correctly render multiple topic tags for news 1`] = `
.emotion-0 {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: -0.0625rem;
  margin-bottom: 0;
  margin-left: -0.25rem;
  margin-right: -0.25rem;
  padding: 0;
}

.emotion-2 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.875rem;
  line-height: 1.125rem;
  word-break: break-word;
  margin-top: 0.0625rem;
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-2 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-2 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

.emotion-2 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #222222;
}

.emotion-2 a:hover,
.emotion-2 a:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2 a:visited {
  color: #6E6E73;
}

.emotion-2 a>span {
  background-color: #F2F2F2;
  padding: 0.375rem 0.4375rem;
}

<div>
  <ul
    class="emotion-0 emotion-1"
    role="list"
  >
    <li
      class="emotion-2 emotion-3"
    >
      <a
        href="#"
      >
        <span>
          test1
        </span>
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        href="#"
      >
        <span>
          test2
        </span>
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        href="#"
      >
        <span>
          test3
        </span>
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        href="#"
      >
        <span>
          test4
        </span>
      </a>
    </li>
  </ul>
</div>
`;

exports[`TopicTags should ignore non-TopicTag children 1`] = `
.emotion-0 {
  list-style-type: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-top: -0.0625rem;
  margin-bottom: 0;
  margin-left: -0.25rem;
  margin-right: -0.25rem;
  padding: 0;
}

.emotion-2 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 0.875rem;
  line-height: 1.125rem;
  word-break: break-word;
  margin-top: 0.0625rem;
  margin-left: 0.25rem;
  margin-right: 0.25rem;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-2 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-2 {
    font-size: 0.8125rem;
    line-height: 1rem;
  }
}

.emotion-2 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #222222;
}

.emotion-2 a:hover,
.emotion-2 a:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-2 a:visited {
  color: #6E6E73;
}

.emotion-2 a>span {
  background-color: #F2F2F2;
  padding: 0.375rem 0.4375rem;
}

<div>
  <ul
    class="emotion-0 emotion-1"
    role="list"
  >
    <li
      class="emotion-2 emotion-3"
    >
      <a
        href="#"
      >
        <span>
          test1
        </span>
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
    >
      <a
        href="#"
      >
        <span>
          test2
        </span>
      </a>
    </li>
  </ul>
</div>
`;
